imagen_pytorch/cli.py

```diff
@@ -50,3 +59,158 @@ def imagen(
+            'timesteps': [1024, 512, 512],
+            'image_sizes': [64, 256, 1024],
+            'random_crop_sizes': [None, 64, 256],
+            'condition_on_text': True,
+            'cond_drop_prob': 0.1,
+            'text_encoder_name': 'google/t5-v1_1-large',
+            'unets': [
+                {
+                    'dim': 512,
+                    'dim_mults': [1, 2, 3, 4],
+                    'num_resnet_blocks': 3,
+                    'layer_attns': [False, True, True, True],
+                    'layer_cross_attns': [False, True, True, True],
+                    'attn_heads': 8
+                },
+                {
+                    'dim': 128,
+                    'dim_mults': [1, 2, 3, 4],
+                    'num_resnet_blocks': [2, 4, 8, 8],
+                    'layer_attns': [False, False, False, True],
+                    'layer_cross_attns': [False, False, False, True],
+                    'attn_heads': 8
+                },
+                {
+                    'dim': 128,
+                    'dim_mults': [1, 2, 3, 4],
+                    'num_resnet_blocks': [2, 4, 8, 8],
+                    'layer_attns': False,
+                    'layer_cross_attns': [False, False, False, True],
+                    'attn_heads': 8
+                }
+            ],
+        },
+        'trainer': {
+            'lr': 1e-4,
+
+        },
+        'dataset_name': 'laion/laion2B-en',
+        'dataset': {
+            'batch_size': 2048,
+            'shuffle': True,
+        },
+        'image_label': None,
+        'url_label': 'URL',
+        'text_label': 'TEXT',
+        'checkpoint_path': './imagen.pt',
+        
+    }
+    with open('./imagen.cfg', 'w') as f:
+        f.write(json.dumps(config, indent = 4))
+
+@imagen.command(help = 'Train the Imagen model')
+@click.option('--config', default = './imagen.cfg', help = 'Path to the Imagen model config')
+@click.option('--unet', default = 1, help = 'Unet to train', type = click.IntRange(1, 3, False, True, True))
+@click.option('--epoches', default = 1000, help = 'Amount of epoches to train for')
+@click.option('--text', required = False, help = 'Text to sample with between epoches', type=str)
+@click.option('--valid', is_flag = True, flag_value=50, default = 0, help = 'Do validation between epoches', show_default = True, type=int)
+def train(
+    config,
+    unet,
+    epoches,
+    text,
+    valid
+):
+    # check config path
+    
+    config_path = Path(config)
+    full_config_path = str(config_path.resolve())
+    assert config_path.exists(), f'config not found at {full_config_path}'
+    
+    with open(config_path, 'r') as f:
+        config_data = json.loads(f.read())
+
+    assert 'checkpoint_path' in config_data, 'checkpoint path not found in config'
+    
+    model_path = Path(config_data['checkpoint_path'])
+    full_model_path = str(model_path.resolve())
+    
+    # setup imagen config
+
+    if config_data['type'] == 'elucidated':
```

Comment:
this could be shortened to

```python
imagen_config_klass = ElucidatedImagenConfig if config_data['type'] == 'elucidated' else ImagenConfig
imagen = imagen_config_klass(**config_data['imagen']).create()
```